config/openiotsdk/cmake/sdk.cmake

```diff
@@ -117,6 +133,17 @@ if(TARGET mbedtls-config)
     )
 endif()
 
+if(TFM_SUPPORT)
+    set(LINKER_SCRIPT ${OPEN_IOT_SDK_CONFIG}/ld/cs300_gcc_tfm.ld)
```

Comment:
Is it possible to check the target being compiled ? Same for `    set(TFM_PLATFORM ${OPEN_IOT_SDK_CONFIG}/tf-m/targets/an552)`. 

It seems these two lines only apply if we build on CS300.
At the same time other assumptions are made about CS300.